src/cmd/manage_nsfs.js

```diff
@@ -406,16 +413,25 @@ async function add_account(data) {
         throw_cli_error(err_code, event_arg, {account: event_arg});
     }
     data._id = mongo_utils.mongoObjectId();
-    const encrypted_account = await nc_mkm.encrypt_access_keys(data);
-    data.master_key_id = encrypted_account.master_key_id;
-    const encrypted_data = JSON.stringify(encrypted_account);
+    data.master_key_id = await nc_mkm.get_active_master_key();
+    let encrypted_data;
+    if (!is_anonymous(data.access_keys)) {
+        const encrypted_account = await nc_mkm.encrypt_access_keys(data);
+        data.master_key_id = encrypted_account.master_key_id;
+        encrypted_data = JSON.stringify(encrypted_account);
+    }
```

Comment:
Can we maybe avoid this condition here and always encrypt?
IIUC we will init() and return the active_master_key even if the array is empty - 
```
async encrypt_access_keys(account) {
        await this.init();
        const master_key_id = this.active_master_key.id;
        const encrypted_access_keys = await P.all(_.map(account.access_keys, async access_keys => ({
            access_key: access_keys.access_key,
            encrypted_secret_key: await this.encrypt(access_keys.secret_key, master_key_id)
        })));
        return { ...account, access_keys: encrypted_access_keys, master_key_id };
    }
```